bread/tests/test_public_visible.py

```diff
@@ -0,0 +1,96 @@
+import random
+import traceback
+
+import django.urls
+from django.contrib.admindocs.views import simplify_regex
+from django.http import HttpResponseRedirect
+from django.template import TemplateDoesNotExist
+from django.test import Client, TestCase
+from django.urls import reverse_lazy as reverse
+from django_extensions.management.commands import show_urls
+
+ROOT_URLPATTERNS = django.urls.get_resolver(None).url_patterns
+ALPHANUMERIC_STR = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
```

Comment:
This is fine like this.
If you do not want to type it by yourself every time there is also the following:

```python
import string
ALPHANUMERIC_STR = string.ascii_letters + string.digits
```

Reply:
That's a good idea. Thank you.